### PR TITLE
Refactoring code

### PR DESCRIPTION
**Pull request by seve-26**
Re-organized code in a more linear structure

### CODE REVIEW

File: vincentvanbot/data.py [hunk 1]
```diff
@@ -26,43 +24,79 @@ def get_data_locally(nrows=10):
...
+    if size:
+        joblib.dump(img_db,FLAT_IMAGES_DB_PATH_ROOT+'_'+str(size)+'.joblib')
+    else:
+        joblib.dump(img_db,FLAT_IMAGES_DB_PATH_ROOT+'.joblib')
+
+def flat_images_db_upload(size=100, rm=False):
+    """Upload df of flat image vectors as joblib file to google cloud"""
+    client = storage.Client().bucket(BUCKET_NAME)
+
+    if size:
```
Comment (berlinikki) [inline vincentvanbot/data.py:56]: Condition 'if size' will be always True, as you have a default value of 100, unless this method is called with size=0. Not sure if this is the assumed behavior. Same for the 'if size' above this piece of code

File: vincentvanbot/data.py [hunk 2]
```diff
@@ -26,43 +24,79 @@ def get_data_locally(nrows=10):
...
+    storage_location = f"data/{local_images_db_name}"
+    blob = client.blob(storage_location)
+    blob.upload_from_filename(FLAT_IMAGES_DB_PATH)
+    print(f"\n=> {local_images_db_name} uploaded to bucket {BUCKET_NAME} inside {storage_location}")
+    if rm:
+        os.remove(FLAT_IMAGES_DB_PATH)
+
+def flat_images_db_download(size=100, source='gcp', rm=True):
+    """Downloads df of flat image vectors as joblib file from source and returns it"""
     if size:
```
Comment (berlinikki) [inline vincentvanbot/data.py:72]: Same here for 'if size'

File: vincentvanbot/data.py [hunk 3]
```diff
@@ -26,43 +24,79 @@ def get_data_locally(nrows=10):
...
-        JOBLIB_PATH = JOBLIB_PATH_ROOT+'.joblib'
-        local_joblib_name = 'flat_resized_images.joblib'
+        FLAT_IMAGES_DB_PATH = FLAT_IMAGES_DB_PATH_ROOT+'.joblib'
+        local_images_db_name = 'flat_resized_images.joblib'
     
     if source == 'local':
-        path = JOBLIB_PATH
+        path = FLAT_IMAGES_DB_PATH
         img_df = joblib.load(path)
     elif source == 'gcp':
```
Comment (berlinikki) [inline vincentvanbot/data.py:82]: I would recommend adding a closing 'else' statement, if there are other possibilities other than 'local' and 'gcp'.

File: vincentvanbot/frompicture/predict.py [hunk 4]
```diff
@@ -1,57 +1,37 @@
...
-        os.remove('train_indexes.joblib')
 
 def get_closest_images_indexes(user_input_transformed, nsimilar=3, rm=True):
     """Takes user_input_transformed as np.array. Downloads fitted knn model and related indexes.
     Returns indexes of nsimilar closest images"""
     client = storage.Client().bucket(BUCKET_NAME)
     
     # download model
-    local_name = 'model_10000.joblib'
+    local_name = 'model.joblib'
```
Comment (rtnikki) [inline vincentvanbot/frompicture/predict.py:17]: Shouldn't we use method 'flat_images_db_download' from data.py here? also this function here doesn't cover the case when size is present in the filename

File: vincentvanbot/labels.py [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+        if label.score > proba_threshold:
+            if manual:
+                if label.description in LABELS_SELECTION:
+                    labels_dict[label.description] = label.score
+            else:
+                labels_dict[label.description] = label.score
+    
+    return labels_dict
+
+def get_labels_from_local_path(path, max_results, proba_threshold=0.6, manual=True):
```
Comment (berlinikki) [inline vincentvanbot/labels.py:42]: A big chunk of code is duplicated from 'get_labels_from_url'. I would suggest refactoring this.

**Review by berlinikki — COMMENTED** [review]
There are a couple of non-critical comments

**Pull request merged**